spark/src/main/scala/geotrellis/spark/io/cog/COGValueReader.scala

```diff
@@ -71,13 +72,69 @@ trait COGValueReader[ID] {
       } catch {
         case th: Throwable => exceptionHandler(key)(th)
       }
+
+    }
+
+    def readSubsetBands(key: K, bands: Seq[Int]): Array[Option[Tile]] = {
```

Comment:
wondering about how we can express `read` using this `readSubsetBands` method.

Reply:
I was thinking something like:

```scala
def read(key: K, bands: Option[Seq[Int]] = None): K
```

Where `None` specifies all bands to be read in. That's still a rough idea, though, so I'd actually need to see if it can be implemented better.

Reply:
Yea, I'm thinking that probably there should be an overload that would convert this `Array[Option[Tile]]` into `V` type.